include/aspect/fluid_pressure_boundary_conditions/interface.h

```diff
@@ -0,0 +1,175 @@
+/*
+  Copyright (C) 2015 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__fluid_pressure_boundary_conditions_interface_h
+#define __aspect__fluid_pressure_boundary_conditions_interface_h
+
+#include <aspect/plugins.h>
+#include <aspect/material_model/interface.h>
+
+namespace aspect
+{
+  /**
+   * A namespace for the definition of things that have to do with describing
+   * the boundary values for fluid pressure for computations with melt
+   * transport.
+   *
+   * @ingroup FluidPressureBoundaryConditions
+   */
+  namespace FluidPressureBoundaryConditions
+  {
+    using namespace dealii;
+
+    /**
+     * Base class
+     *
+     * @ingroup FluidPressureBoundaryConditions
+     */
+    template <int dim>
+    class Interface
+    {
+      public:
+        /**
+         * Destructor. Made virtual to enforce that derived classes also have
+         * virtual destructors.
+         */
+        virtual ~Interface();
+
+        /**
+         * Initialization function. This function is called once at the
+         * beginning of the program after parse_parameters is run and after the
+         * SimulatorAccess (if applicable) is initialized.
+         */
+        virtual void initialize ();
+
+
+        /**
+         * Compute the gradient of the fluid pressure
+         * for each quadrature point.
+         *
+         * The return value can typically contain @p material_model_outputs.fluid_densities[q]
+         * or @p material_model_outputs.densities[q], multiplied by the gravity vector.
+         *
+         * @param material_model_inputs The material property inputs.
+         * @param material_model_inputs The material property outputs.
+         * @param output Result to be filled.
+         */
+        virtual
+        void fluid_pressure_gradient (
+          const MaterialModel::MaterialModelInputs<dim> &material_model_inputs,
+          const MaterialModel::MaterialModelOutputs<dim> &material_model_outputs,
+          std::vector<Tensor<1,dim> > &output
```

Comment:
Another thought: For all other boundary conditions we have added the boundary_id as additional argument to this function. Please do that here as well in case somebody wants to write an 'Ascii data' fluid pressure plugin. Otherwise we run in the same deprecation problem as for the other boundary condition plugins.